### PR TITLE
Add current user and new user creation form.

### PR DESCRIPTION
**Pull request by flibbertigibbet**

### CODE REVIEW

File: src/nih_wayfinding/app/scripts/profiles/profile-service.js [hunk 1]
```diff
@@ -9,12 +12,49 @@
...
+            }
+        }
+
+        /**
+         * Sets a property on the currently active user and saves it to local storage.
+         *
+         * @param property {String} Name of the property to set
+         * @param value {Object} Value to set the property to
+         */
+         function setUserProperty(property, value) {
```
Comment (CloudNiner) [inline src/nih_wayfinding/app/scripts/profiles/profile-service.js]: Might be better named as 'setCurrentUserProperty'. When I first saw the function signature I assumed I could set a property for any user.

File: src/nih_wayfinding/app/scripts/views/profile/new-profile/new-profile-partial.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<div class="content scroll">
+    <div class="panel panel-default">
+        <div class="panel-heading">
+            <h3 class="panel-title">Create New User {{ newprofile.displayUsername }}</h3>
+            <p>Question {{ newprofile.step }}/3</p>
+        </div>
+        <div class="panel-body">
+            <div ng-switch="newprofile.step">
+                <div ng-switch-when="1">
```
Comment (CloudNiner) [inline src/nih_wayfinding/app/scripts/views/profile/new-profile/new-profile-partial.html:9]: Nice way to do this!

**Comment by CloudNiner** [comment]
I've been defining the private vars and functions in these modules in alphabetical order for consistency/readability/searchability, and was going to add this to the style guide. Thoughts?

Ping to @moradology as well since he's working in this codebase now.

**Comment by flibbertigibbet** [comment]
I generally try to list together variable and function definitions that logically belong together.  I don't know if there's benefit to be gained from alphabetizing them, and a not inconsiderable amount of extra work to go into enforcing such a rule.  Also, I don't think I've ever seen another codebase that does that.  If a given file has so much stuff in it that it needs that level of organization, it would probably be better off getting refactored into smaller pieces anyway.

**Comment by flibbertigibbet** [comment]
If the team is going to have a discussion about inventing new style rules, maybe a PR isn't the best place to do that.

**Comment by CloudNiner** [comment]
Yeah, that's fine for now. I didn't have a strong opinion about it. I just dislike when functions end up defined in willy-nilly order and there's no structure at all.

**Comment by CloudNiner** [comment]
+1

**Comment by flibbertigibbet** [comment]
The definition ordering doesn't look logical to you here?  The service functions are grouped by the field they work on, with getters and setters together.  The controller functions are defined in the order of their related steps, above the one validator, and the initialization function is at the end, where we always put it in controllers.

**Comment by CloudNiner** [comment]
No, sorry, here looks good. I mean that things can become a bit hectic after a few revisions where modifications get made a bit at a time and all of a sudden functions are listed in apparently random order.

**Comment by flibbertigibbet** [comment]
Right.  Yes, that does happen.

**Pull request merged**